src/app/layout/add-contact/name-validator.ts

```diff
@@ -0,0 +1,10 @@
+import { AbstractControl, ValidationErrors } from '@angular/forms';  
+    
+export class NameValidator {  
+    static cannotContainSpace(control: AbstractControl) : ValidationErrors | null {  
+        if((control.value as string).indexOf(' ') >= 0){  
```

Comment:
Just be aware with this that, if you were to put this validator on a `<input type="number" />` that it would cause an error.  `as string` doesn't cast the variable to a string, just makes typescript think it is a string.